core/templates/components/summary-tile/collection-summary-tile.component.ts

```diff
@@ -30,24 +30,26 @@ import { UserService } from 'services/user.service';
   templateUrl: './collection-summary-tile.component.html',
 })
 export class CollectionSummaryTileComponent implements OnInit {
-  @Input() getCollectionId: string;
-  @Input() getCollectionTitle: string;
-  @Input() getLastUpdatedMsec: number;
-  @Input() getObjective: string;
-  @Input() getNodeCount: string;
-  @Input() getCategory: string;
-  @Input() getThumbnailIconUrl: string;
-  @Input() getThumbnailBgColor: string;
-  @Input() isPlaylistTile: boolean;
-  @Input() isLinkedToEditorPage: boolean;
-  @Input() showLearnerDashboardIconsIfPossible: string;
-  @Input() isContainerNarrow: boolean;
-  @Input() isOwnedByCurrentUser: boolean;
+  // These properties are initialized using component interactions,
+  // see https://github.com/oppia/oppia/wiki/Guide-on-defining-types#ts-7-1.
+  @Input() getCollectionId!: string;
+  @Input() getCollectionTitle!: string;
+  @Input() getLastUpdatedMsec!: number;
+  @Input() getObjective!: string;
+  @Input() getNodeCount!: string;
+  @Input() getCategory!: string;
+  @Input() getThumbnailIconUrl!: string;
+  @Input() getThumbnailBgColor!: string;
+  @Input() isPlaylistTile: boolean = false;
+  @Input() isLinkedToEditorPage: boolean = false;
+  @Input() showLearnerDashboardIconsIfPossible!: string;
+  @Input() isContainerNarrow: boolean = false;
+  @Input() isOwnedByCurrentUser: boolean = false;
 
-  userIsLoggedIn: boolean;
-  collectionIsCurrentlyHoveredOver: boolean;
-  defaultEmptyTitle: string;
-  activityTypeCollection: string;
+  userIsLoggedIn: boolean = false;
+  collectionIsCurrentlyHoveredOver: boolean = false;
+  defaultEmptyTitle!: string;
```

Comment:
Are these two properties being initialized in angular lifecycle hooks? If not then how are we sure they will always have a value whenever used?

Reply:
But what will leaving them undefine mean here? wouldn't they still be false?

Reply:
Would `defaultEmptyTitle` and `activityTypeCollection` still be false? Not quite sure what you mean here.

Reply:
But I have not Initialized these two properties right?

Reply:
Using **(!)** means that we are asserting to the compiler that these properties will be initialized somewhere even if the compiler itself can not figure this out. When we used the non-null assertion operator on properties like `@input() getCategory` , the compiler could not tell that these properties are being initialized in angular lifecycle hooks so it gave out an error that these need to be initialized. Does that make sense? You would need a comment explaining where these two properties are initialized or add `|undefined` and place checks in the code.

Reply:
Ah ok! These are populated using ngOnInit added a comment PTAL